debuginfo/src/dwarf.rs

```diff
@@ -553,21 +552,48 @@ impl<'d, 'a> DwarfUnit<'d, 'a> {
             let rows = line_program.get_rows(range);
             lines.reserve(rows.len());
 
-            for row in rows {
-                let file = self.resolve_file(row.file_index).unwrap_or_default();
-                let line = row.line.unwrap_or(0);
+            if let Some((first, rows)) = rows.split_first() {
+                let file = self.resolve_file(first.file_index).unwrap_or_default();
+                let line = first.line.unwrap_or(0);
+                let size = first.size.map(|s| s + first.address - range.begin);
```

Comment:
Can we reduce the code duplication here? If I'm reading this correctly, then there are two changes in this:
 1. `last` is moved into the loop around `ranges`. 
 2. You patch line size of the previous record. You could probably do that essentially in every iteration when you have a new record, before the `continue` block.

Reply:
Discussed offline: Could you also add a comment that explains why the first row is corrected to align with the start of the `range`.